cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -360,52 +362,47 @@ def get_full_psd_matrix(self):
         axis=0)
     return self.matrix_h, self.matrix_m
 
-  def make_m_psd(self, feed_dict):
+  def make_m_psd(self, original_nu, feed_dict):
```

Comment:
This and few other functions modify `feed_dict` which could be used somewhere else, ex:
```
feed_dict = { .... }
make_m_psd(nu, feed_dict)
sess.run(something_else, feed_dict=feed_dict)
```

As a result you have to track what is in feed_dict and what was modified. Also it may lead to hard to find bugs when function modified feed_dict while you didn't expect this.

So I would recommend to make a copy of feed dict if function is intended to modify it:
```
def make_m_psd(self, original_nu, feed_dict):
  feed_dict = feed_dict.copy()
  ...
```

Please check the rest of the code it applies to other functions as well